thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/auth/ThirdEyeAuthenticatorLdap.java

```diff
@@ -29,48 +36,155 @@ public ThirdEyeAuthenticatorLdap(String domainSuffix, String ldapUrl) {
    */
   @Override
   public Optional<ThirdEyePrincipal> authenticate(Credentials credentials) throws AuthenticationException {
-    String principalName = credentials.getPrincipal();
     try {
-      if (principalName != null) {
-        // Append the default domain name if the given principal name doesn't have one
-        if (!principalName.contains("@")) {
-          principalName = principalName + '@' + this.domainSuffix;
-        }
-        LOG.info("Authenticating {} via username and password", principalName);
+      String principalName = credentials.getPrincipal();
+      if (StringUtils.isBlank(principalName)) {
+        LOG.info("Unable to authenticate empty user name.");
+        return Optional.absent();
+      } else {
+        LOG.info("Authenticating '{}' via username and password", principalName);
 
         Hashtable<String, String> env = new Hashtable<>();
         env.put(Context.INITIAL_CONTEXT_FACTORY, LDAP_CONTEXT_FACTORY);
         env.put(Context.PROVIDER_URL, this.ldapUrl);
-
         if (this.ldapUrl.startsWith("ldaps")) {
           env.put(Context.SECURITY_PROTOCOL, "ssl");
         }
         env.put(Context.SECURITY_AUTHENTICATION, "simple");
-        env.put(Context.SECURITY_PRINCIPAL, principalName);
         env.put(Context.SECURITY_CREDENTIALS, credentials.getPassword());
 
-        // Attempt ldap authentication
-        try {
-          new InitialDirContext(env).close();
-        } catch (NamingException e) {
-          LOG.error("Could not authenticate {} with LDAP", principalName, e);
+        // Attempt ldap authentication with the following steps:
+        // 1. If user's name contains domain name or the system doesn't have any given domain names, then
+        //    use the username as is.
+        // 2. Else, try out all combinations of username and the given domain names of the system.
+        AuthenticationResults authenticationResults = new AuthenticationResults();
+        if (principalName.contains("@") || CollectionUtils.isEmpty(domainSuffix)) {
+          env.put(Context.SECURITY_PRINCIPAL, principalName);
+          AuthenticationResult authenticationResult = authenticate(env);
+          authenticationResults.appendAuthenticationResult(authenticationResult);
+        } else {
+          for (String suffix : domainSuffix) {
+            env.put(Context.SECURITY_PRINCIPAL, principalName + '@' + suffix);
+            AuthenticationResult authenticationResult = authenticate(env);
+            authenticationResults.appendAuthenticationResult(authenticationResult);
+            if (authenticationResults.isAuthenticated()) {
+              break;
+            }
+          }
+        }
+
+        if (authenticationResults.isAuthenticated()) {
+          ThirdEyePrincipal principal = new ThirdEyePrincipal();
+          principal.setName(env.get(Context.SECURITY_PRINCIPAL));
+          LOG.info("Successfully authenticated {} with LDAP", principalName);
```

Comment:
will be good to know which one was used to authenticate

Reply:
See https://github.com/linkedin/pinot/pull/2591